org.civicrm.reqangular/test/decorators/ui-bootstrap/uib-modal.decorator.spec.js

```diff
@@ -0,0 +1,83 @@
+/* eslint-env amd, jasmine */
+
+define([
+  'common/lodash',
+  'common/angularMocks',
+  'common/angular',
+  'common/decorators/ui-bootstrap/uib-modal.decorator',
+  'common/angularBootstrap'
+], function (_, angularMocks, angular, $uibModalDecorator, $document) {
+  'use strict';
+
+  describe('$uibModal.open', function () {
+    var $uibModal, $provide, $document;
+
+    beforeEach(module('ui.bootstrap'));
+
+    beforeEach(function () {
+      module(function (_$provide_) {
+        $provide = _$provide_;
+      });
+      inject(function () {
+        $provide.decorator('$uibModal', $uibModalDecorator);
+      });
+    });
+
+    beforeEach(inject(function (_$uibModal_, _$document_) {
+      $uibModal = _$uibModal_;
+      $document = _$document_;
+    }));
+
+    describe('init', function () {
+      var arg1 = { template: '<component></component>' };
+      var arg2 = 'another_argument';
+      var elements = {};
+      var doc, originalStyles;
+
+      beforeEach(function () {
+        doc = $document[0];
+        elements.body = doc.body;
+        elements.html = doc.getElementsByTagName('html')[0];
+
+        if (!originalStyles) {
```

Comment:
why do you need to check if the var is already assigned?

Reply:
I need to assign in only **once**. I tried beforeAll, but it is executed **after** nested beforeEach.